tests/integration/targets/nd_federation_member/tasks/main.yml

```diff
@@ -0,0 +1,133 @@
+
+# - name: Test that we have an Nexus Dashboard Insights Group and Site Name defined
+#   fail:
+#     msg: "Please define the following variables: insights_group, site_name."
+#   when: insights_group is not defined or site_name is not defined
+
+# SET VARS
+- name: Set vars
+  ansible.builtin.set_fact:
+    nd_info: &nd_info
+      output_level: '{{ site_output_level | default("debug") }}'
+
+
+# CLEAN ENVIRONMENT
+- name: Delete federation
+  cisco.nd.nd_federation:
+    <<: *nd_info
+    name: 'lh-dmz1-pod1-ndo-v402'
+    state: absent
+
+# ADD Federation
+- name: Add ND federation
+  cisco.nd.nd_federation: &add_federation
+    <<: *nd_info
+    name: 'lh-dmz1-pod1-ndo-v402'
+    state: present
+  register: add_federation
+
+- name: Assertion check for adding ND federation 
+  ansible.builtin.assert:
+    that:
+      - add_federation is changed
+      - add_federation.previous is none
+      - add_federation.current.spec.name == 'lh-dmz1-pod1-ndo-v402'
+
+# ADD Federation member (multicluster setup)
+- name: Add ND federation member (check mode)
+  cisco.nd.nd_federation_member: &attach_federation_member
+    <<: *nd_info
+    cluster: '173.36.219.33'
+    cluster_username: 'admin'
+    cluster_password: 'ins3965!'
+    state: present
+  check_mode: True
+  register: cm_add_federation_member
+
+- name: Add ND federation member (normal mode)
+  cisco.nd.nd_federation_member:
+    <<: *attach_federation_member
+    state: present
+  register: nm_add_federation_member
+
+- name: Add ND federation member again
+  cisco.nd.nd_federation_member:
+    <<: *attach_federation_member
+    state: present
+  register: add_federation_member_again
+
+- name: Assertion check for attaching ND federation member 
+  ansible.builtin.assert:
+    that:
+      - cm_add_federation_member is changed
+      - nm_add_federation_member is changed
+      - add_federation_member_again.current.spec.host == '173.36.219.33'
+      - cm_add_federation_member.previous == nm_add_federation_member.previous == {}
+      - cm_add_federation_member.current.spec.host == nm_add_federation_member.current.spec.host == '173.36.219.33'
+      - cm_add_federation_member.current.spec.loginDomain == nm_add_federation_member.current.spec.loginDomain == 'DefaultAuth'
+      - cm_add_federation_member.current.spec.userName == nm_add_federation_member.current.spec.userName == 'admin'
+
+# Query Federation members (multicluster setup)
+- name: Query all federation members
+  cisco.nd.nd_federation_member:
+    <<: *nd_info
+    state: query
+  register: query_all_federation_members
+
+- name: Query a federation member
+  cisco.nd.nd_federation_member:
+    <<: *nd_info
+    cluster: '173.36.219.33'
+    state: query
+  register: query_federation_member
+
+- name: Assertion check for querying ND federation members
+  ansible.builtin.assert:
+    that:
+    - query_all_federation_members is not changed
```

Comment:
should we add another member and should we check expected length (amount of member)?

Reply:
added